app/updatemanager_test.go

```diff
@@ -576,3 +576,358 @@ func TestActiveAndExpiredPoolKeys(t *testing.T) {
 		4*time.Second,
 		500*time.Millisecond)
 }
+
+func TestQueryLogic(t *testing.T) {
+	active := 3
+	expired := 0
+	t.Run("Fail exists: Action", func(t *testing.T) {
```

Comment:
All these could probably be written into a nice table test, if you find them hard to decipher